### PR TITLE
Add authentication flow, users and resolutions - basic skeleton

### PR DESCRIPTION
**Pull request by ypicard**
- **Authentication** flow working (login + logout) via JWT tokens
- **Signup** flow working
- Usage of **InheritedWidget** (cf. https://flutterbyexample.com, Auth part) to have a top level appState which will show login screen or home page depending on if a user is authenticated or not
- Store jwt token in **shared preferences**
- JWT parsed using https://stackoverflow.com/questions/52017389/how-to-get-the-claims-from-a-jwt-in-my-flutter-application#52021206 because the jwt_dart library does not support dart2
- **Globals** file to store useful vars
- **Resolution** model
- Create resolution flow, and add it to current user
- List user resolutions
- View specific resolution

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,190 +1,95 @@
...
   }
-}
 
-class Signup extends StatefulWidget {
-  @override
-  SignupState createState() {
-    return SignupState();
+  Widget _buildResolutionList() {
+    _setUser();
+    List<Resolution> resolutions = _user == null ? [] : _user.resolutions;
```
Comment (sebastian-sierra) [inline lib/main.dart:73]: I find the name of this function a little bit misleading. When I see the signature `Widget _buildResolutionList() ` it does not come to me that we might modify the state of the widget by calling `_setUser()`. 
This should only return a widget and leave this state as it is (I do not know exactly the implementation details of flutter, but in react, modifying the state in a render method is a no-no. See: https://stackoverflow.com/questions/48226268/calling-setstate-in-react-from-render-method).

Maybe we can do this in the lifecycle hook method of a widget `initState`(https://flutterbyexample.com/stateful-widget-lifecycle/#3-initstate).

File: lib/models/test.dart [hunk 2]
```diff
@@ -0,0 +1 @@
+class Test {}
```
Comment (sebastian-sierra) [inline lib/models/test.dart]: Delete unused(?)

File: lib/pages/settings_page.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    _getUser();
+    return Scaffold(
+        appBar: AppBar(
+          title: Text('Settings'),
+        ),
+        key: _scaffoldKey,
+        body: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            _user == null ? null : Text("First name: " + _user.firstName),
```
Comment (sebastian-sierra) [inline lib/pages/settings_page.dart:29]: We can try to avoid the duplicated condition

Comment (ypicard) [reply lib/pages/settings_page.dart:29]: I don't think there is a simpler way of checking null on the go in dart in this case (nothing like `_user ? x : y` in javascript). 
But this code will be deleted later on, when a proper layout will be built for this page anyways.

File: lib/models/test.dart (hunk 2 above)
Comment (ypicard) [reply lib/models/test.dart]: Ok

File: lib/main.dart (hunk 1 above)
Comment (ypicard) [reply lib/main.dart:73]: Called in `build` hook: `appState` isn't defined yet in `initState` hook.

**Pull request closed without merging**